### PR TITLE
clippy: Fix various clippy problems in `components/scripts/dom`

### PR DESCRIPTION
**Pull request by six-shot**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are part of #31500.
- [x] These changes do not require tests because they do not modify functionality.

### CODE REVIEW

File: components/script/dom/globalscope.rs [hunk 1]
```diff
@@ -294,15 +294,15 @@ pub struct GlobalScope {
     ///
     /// <https://html.spec.whatwg.org/multipage/#about-to-be-notified-rejected-promises-list>
     #[ignore_malloc_size_of = "mozjs"]
-    uncaught_rejections: DomRefCell<Vec<Box<Heap<*mut JSObject>>>>,
```
Comment (mrobinson) [inline components/script/dom/globalscope.rs:297]: What was the error that led you to removing the `Box` here?

Comment (six-shot) [reply components/script/dom/globalscope.rs:297]: It was saying box is unnecessary 
https://rust-lang.github.io/rust-clippy/master/index.html#/vec_box

File: components/script/dom/globalscope.rs [hunk 2]
```diff
@@ -2179,42 +2179,42 @@ impl GlobalScope {
     pub fn add_uncaught_rejection(&self, rejection: HandleObject) {
         self.uncaught_rejections
             .borrow_mut()
-            .push(Heap::boxed(rejection.get()));
+            .push(*Heap::boxed(rejection.get()));
```
Comment (mrobinson) [inline components/script/dom/globalscope.rs:2182]: This is creating a `Box<Heap<>>` and then immediately pulling it apart. I would either keep using `Box<...>` in the vector or figure out a way to create this `Heap` without putting it in a `Box<...>`.

File: components/script/dom/globalscope.rs (hunk 1 above)
Comment (mrobinson) [reply components/script/dom/globalscope.rs:297]: I think for now you should just allow the warning for `uncaught_rejections` and `consumed_rejections` below. It seems that `Heap` always comes boxed and it's tricky to undo that. Instead put a comment like:

```rust
/// TODO: `Vec` already stores members in a `Box` so this extra box is an unnecessary layer of
/// indirection. Can it be removed?
```

cc @sagudev

**Review by mrobinson — CHANGES_REQUESTED** [review]

**Review by mrobinson — APPROVED** [review]

**Pull request merged**